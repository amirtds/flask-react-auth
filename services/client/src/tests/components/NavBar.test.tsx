import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { describe, it, expect, vi, afterEach } from "vitest";
import "@testing-library/jest-dom/vitest";

import NavBar from "../../components/NavBar";

describe("NavBar", () => {
  afterEach(() => {
    cleanup();
    vi.restoreAllMocks();
  });

  const mockLogoutUser = vi.fn();

  it("NavBar renders without crashing", () => {
    render(
      <Router>
        <NavBar
          title="Hello, World!"
          logoutUser={mockLogoutUser}
          isAuthenticated={() => false}
        />
      </Router>,
    );

    const titleElement = screen.getByText("Hello, World!");
    expect(titleElement).toBeInTheDocument();
    expect(titleElement.closest("h1")).toHaveClass("navbar-item");
    expect(titleElement.closest("h1")).toHaveClass("nav-title");
  });

  it("NavBar contains correct navigation links when user is logged out", () => {
    render(
      <Router>
        <NavBar
          title="Hello, World!"
          logoutUser={mockLogoutUser}
          isAuthenticated={() => false}
        />
      </Router>,
    );

    // Links that should be visible when logged out
    expect(screen.getByRole("link", { name: "About" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Register" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Log In" })).toBeInTheDocument();

    // Links that should be hidden when logged out
    expect(
      screen.queryByRole("link", { name: "User Status" }),
    ).not.toBeInTheDocument();
    expect(screen.queryByText("Log Out")).not.toBeInTheDocument();
  });

  it("NavBar contains correct navigation links when user is logged in", () => {
    render(
      <Router>
        <NavBar
          title="Hello, World!"
          logoutUser={mockLogoutUser}
          isAuthenticated={() => true}
        />
      </Router>,
    );

    // Links that should be visible when logged in
    expect(screen.getByRole("link", { name: "About" })).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: "User Status" }),
    ).toBeInTheDocument();

    // Check for Log Out link
    const logOutLinks = screen.getAllByText("Log Out");
    expect(logOutLinks.length).toBeGreaterThan(0);
    expect(logOutLinks[0]).toBeInTheDocument();

    // Links that should be hidden when logged in
    expect(
      screen.queryByRole("link", { name: "Register" }),
    ).not.toBeInTheDocument();
    expect(
      screen.queryByRole("link", { name: "Log In" }),
    ).not.toBeInTheDocument();
  });

  it("NavBar renders properly", () => {
    const { asFragment } = render(
      <Router>
        <NavBar
          title="Hello, World!"
          logoutUser={mockLogoutUser}
          isAuthenticated={() => false}
        />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
