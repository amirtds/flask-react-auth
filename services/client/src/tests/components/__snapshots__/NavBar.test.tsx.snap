// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar renders properly 1`] = `
<DocumentFragment>
  <div
    class="css-tqkzpo"
  >
    <div
      class="css-q4hjtx"
    >
      <div
        class="css-70qvj9"
      >
        <h1
          class="chakra-heading navbar-item nav-title css-15ro776"
        >
          <a
            class="chakra-link css-3yqgza"
            href="/"
          >
            Hello, World!
          </a>
        </h1>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-1du6rpu"
      >
        <a
          class="chakra-link css-b80v07"
          href="/about"
        >
          About
        </a>
        <a
          class="chakra-link css-b80v07"
          href="/status"
        >
          User Status
        </a>
        <a
          class="chakra-link css-b80v07"
          href="/register"
        >
          Register
        </a>
        <a
          class="chakra-link css-b80v07"
          href="/login"
        >
          Log In
        </a>
      </div>
      <button
        aria-label="Toggle Navigation"
        class="chakra-button css-8lrqwb"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="chakra-collapse"
      style="overflow: hidden; display: none; opacity: 0; height: 0px;"
    >
      <div
        class="chakra-stack css-p0xbc3"
      >
        <a
          class="chakra-link css-b80v07"
          href="/about"
        >
          About
        </a>
        <a
          class="chakra-link css-b80v07"
          href="/status"
        >
          User Status
        </a>
        <a
          class="chakra-link css-b80v07"
          href="/register"
        >
          Register
        </a>
        <a
          class="chakra-link css-b80v07"
          href="/login"
        >
          Log In
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
